src/components/switch.tsx

```diff
@@ -0,0 +1,47 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import React, { forwardRef } from 'react'
+
+import { cn } from '@/utils/cn'
+
+const switchVariants = cva(
+  [
+    'rounded-full bg-disabled-strong relative cursor-pointer transition-all duration-300 ease-in-out',
+    'before:content-[""] before:absolute before:bg-white before:rounded-full',
+    'before:top-1/2 before:transform before:-translate-y-1/2 before:left-[4px]',
+    'before:transition-all before:duration-300 before:ease-in-out',
+    'peer-checked:before:left-[18px] peer-checked:bg-switch-base',
```

Comment:
Let's add the focus state as well:
```suggestion
    'peer-checked:before:left-[18px] peer-checked:bg-switch-base',
    'peer-focus:outline peer-focus:outline-2 peer-focus:outline-blue-500',
```